sdk/core/Azure.Core/tests/TransportFunctionalTests.cs

```diff
@@ -917,6 +920,65 @@ public async Task StreamReadingExceptionsAreIOExceptions()
             }
         }
 
+        [Test]
+        public async Task ServerCertificateCustomValidationCallbackIsHonored([Values(true, false)] bool setCertCallback,[Values(true, false)]  bool isValidCert)
+        {
+            using (TestServer testServer = new TestServer(
+                async context =>
+                {
+                    byte[] buffer = Encoding.UTF8.GetBytes("Hello");
+                    await context.Response.Body.WriteAsync(buffer, 0, buffer.Length);
+                },
+                true))
+            {
+                bool certValidationCalled = false;
+                var options = new HttpPipelineTransportOptions();
+
+                if (setCertCallback)
+                {
+                    options.ServerCertificateCustomValidationCallback = certificate2 =>
+                    {
+                        certValidationCalled = true;
+                        Assert.NotNull(certificate2);
+                        return isValidCert;
+                    };
+                }
+                var transport = GetTransport(true, options);
+                Request request = transport.CreateRequest();
+                request.Uri.Reset(testServer.Address);
+
+                try
+                {
+                    await ExecuteRequest(request, transport);
+                    if (setCertCallback)
```

Comment:
We can only get here with setCertCallback == true && isValidCert == true


Assert.IsTrue(isValidCert);
Assert.IsTrue(setCertCallback);

Reply:
On full framework, that is not true because we were setting the global ServicePoint handler.  Fixing that.